app/lib/authentication/strategy/base.rb

```diff
@@ -22,13 +22,19 @@ def initialize(strategy, users, params, user_data)
     class Base
       include System::UrlHelpers.cms_url_helpers
 
-      attr_reader :site_account, :admin_domain, :user_for_signup, :new_user_created
+      attr_reader :site_account, :admin_domain, :user_for_signup, :new_user_created, :error_message
+
+      def self.expected_params
+        raise NotImplementedError, "expected #{self} to implement #{__method__}"
+      end
 
       def initialize(site_account, admin_domain = false)
         @site_account     = site_account
         @admin_domain     = admin_domain
         @user_for_signup  = nil
         @new_user_created = false
+
+        Rails.logger.debug("Trying to log in by #{name} auth strategy") unless is_a?(Authentication::Strategy::Null)
```

Comment:
This is an anti-pattern to add such checks. Just override `#initialize` in `Authentication::Strategy::Null` to do nothing.

Reply:
I don't want to say you were right, so I won't say it. https://github.com/3scale/porta/pull/4064/commits/6a156e25d1353fc048209ff44af06dffd9d9d553